pkg/spoke/spokeagent.go

```diff
@@ -282,14 +273,26 @@ func (o *SpokeAgentOptions) RunSpokeAgent(ctx context.Context, controllerContext
 	go managedClusterJoiningController.Run(ctx, 1)
 	go managedClusterLeaseController.Run(ctx, 1)
 	go managedClusterHealthCheckController.Run(ctx, 1)
-	go managedClusterClaimController.Run(ctx, 1)
+	if features.DefaultMutableFeatureGate.Enabled(features.ClusterClaim) {
+		// create managedClusterClaimController to sync cluster claims
+		managedClusterClaimController := managedcluster.NewManagedClusterClaimController(
+			o.ClusterName,
+			o.MaxCustomClusterClaims,
+			hubClusterClient,
+			hubClusterInformerFactory.Cluster().V1().ManagedClusters(),
+			spokeClusterInformers.Cluster().V1alpha1().ClusterClaims(),
```

Comment:
declaring  informers from factory after factory is started might be a problem.

Reply:
good catch, it seems that cluster-claim controller is the only one that actually references `spokeClusterInformers`, so 
i wrapped that into the feature-gated block.